feat: improvments on decryption (#33513)
mdn/content: files/en-us/glossary/decryption/index.md

{{GlossarySidebar}}

In {{glossary("cryptography")}}, **decryption** is the conversion of {{glossary("ciphertext")}} into {{glossary("plaintext")}}.

Decryption is an operation which transforms a ciphertext message into plaintext using a cryptographic algorithm called a {{glossary("cipher")}}. Like encryption, decryption in modern ciphers is performed by using a specific algorithm and a secret, called the {{glossary("key")}}.

![The decryption primitive.](decryption.png)

> Decryption is the reverse process of {{glossary("encryption")}} and if the key stays secret, is mathematically hard to perform. The difficulty depends on the security of the cryptographic algorithm chosen and the research progress of {{glossary("cryptanalysis")}}.

Reviewer: > in the original context, evolves should means as cryptanalysis develops?

Yes, I think so.

> so can we just say, the progress of that development(research)?

```suggestion
Decryption is the reverse process of {{glossary("encryption")}} and if the key stays secret, is mathematically hard to perform. How hard it is depends on how secure the cryptographic algorithm is, and that in itself is subject to change as the study of {{glossary("cryptanalysis")}} advances.
```

> In {{glossary("cryptography")}}, **decryption** is the conversion of {{glossary("ciphertext")}} into {{glossary("Plaintext")}}.

Reviewer: ```suggestion
In {{glossary("cryptography")}}, **decryption** is the conversion of {{glossary("ciphertext")}} into {{glossary("plaintext")}}.
```

> Decryption is a cryptographic primitive: it transforms a ciphertext message into plaintext using a cryptographic algorithm called a {{glossary("cipher")}}. Like encryption, decryption in modern ciphers is performed by using a specific algorithm and a secret, called the {{glossary("key")}}. Since the algorithm is often public, the key must stay secret to ensure that the encryption is secure.

Reviewer: ```suggestion
Decryption is an operation which transforms a ciphertext message into plaintext using a cryptographic algorithm called a {{glossary("cipher")}}. Like encryption, decryption in modern ciphers is performed by using a specific algorithm and a secret, called the {{glossary("key")}}.
```

- I don't think "cryptographic primitive" is helpful jargon here.
- I don't think that "Since the algorithm is often public..." sentence is helpful (and is potentially misleading, wrongly implying that if the algorithm is not public, then the key need not be secret).

Author: - u're right. it's not helpful (not straight-forward).
- yeap, having the same idea here why the editor has this line added.

![The decryption primitive.](decryption.png)

> Decryption is the reverse process of {{glossary("encryption")}} and if the key stays secret, decryption without knowing the specific secret, decryption is mathematically hard to perform, depending on the security of the cryptographic algorithm chosen and evolves with the progress of {{glossary("cryptanalysis")}}.

Reviewer: ```suggestion
Decryption is the reverse process of {{glossary("encryption")}} and if the key stays secret, is mathematically hard to perform. How hard depends on the security of the cryptographic algorithm chosen and evolves with the progress of {{glossary("cryptanalysis")}}.
```

- there was another error in this sentence, " decryption without knowing the specific secret"
- I think the original 2-sentence form was fine, and the PR introduced a new error ("evolves" no longer made sense).

Author: in the original context, `evolves` should means as cryptanalysis develops? so can we just say, the progress of that develpment(research)?

```suggestion
Decryption is the reverse process of {{glossary("encryption")}} and if the key stays secret, is mathematically hard to perform. The difficulty depends on the security of the cryptographic algorithm chosen and the research progress of {{glossary("cryptanalysis")}}.
```
